### PR TITLE
Add Wiz plugin initial tables, docs. Closes #1

### LINKED ISSUE #1: Add Wiz plugin initial tables, docs

### PR DESCRIPTION
**Pull request by Subhajit97**
- New tables added:
  - `wiz_cloud_config_rule`
  - `wiz_cloud_configuration_finding`
  - `wiz_control`
  - `wiz_issue`
  - `wiz_project`
  - `wiz_security_framework`
  - `wiz_service_account`
  - `wiz_subscription`
  - `wiz_user`
  - `wiz_user_role`
  - `wiz_vulnerability_finding`

### CODE REVIEW

File: docs/tables/wiz_vulnerability_finding.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Table: wiz_vulnerability_finding
+
+The `wiz_vulnerability_finding` table can be used to list all vulnerabilities that Wiz scanner detected on the cloud resources.
+
+**Note**: The table can limit a large dataset; which can increase the query execution time. It is recommended that queries to this table should include (usually in the `where` clause) at least one of these columns:
```
Comment (misraved) [inline docs/tables/wiz_vulnerability_finding.md:5]:
```suggestion
**Note**: The table can return a large dataset; which can increase the query execution time. It is recommended that queries to this table should include (usually in the `where` clause) at least one of these columns:
```

File: docs/tables/wiz_vulnerability_finding.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  has_exploit,
+  vulnerable_asset ->> 'subscriptionId' as subscription
+from
+  wiz_vulnerability_finding
+where
+  has_exploit
+  and severity = 'HIGH';
+```
+
+### List high severity findings which are fixed
```
Comment (misraved) [inline docs/tables/wiz_vulnerability_finding.md:51]: I think it is better to add a query about high severity findings that are not fixed

File: wiz/service.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+	return &at, nil
+}
+
+/*
+Returns credentials by precedence.
+
+Precedence of credentials:
+  - Credentials set in config
+  - Value set using WIZ_AUTH_CLIENT_ID, WIZ_AUTH_CLIENT_SECRET, and WIZ_URL env var
```
Comment (misraved) [inline wiz/service.go:156]:
```suggestion
  - Value set using WIZ_AUTH_CLIENT_ID, WIZ_AUTH_CLIENT_SECRET, and WIZ_URL env variables
```

File: wiz/table_wiz_cloud_config_rule.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+			{Name: "id", Type: proto.ColumnType_STRING, Description: "A unique identifier of the cloud configuration rule."},
+			{Name: "enabled", Type: proto.ColumnType_BOOL, Description: "Rule enabled status."},
+			{Name: "severity", Type: proto.ColumnType_STRING, Description: "Rule severity will outcome to finding severity. This filed initial value is set as the severity of the CSPM rule. Possible values are: CRITICAL, HIGH, INFORMATIONAL, LOW, MEDIUM."},
+			{Name: "cloud_provider", Type: proto.ColumnType_STRING, Description: "The cloud provider this rule is relevant to. Possible values are: AWS, GCP, OCI, Alibaba, Azure, Kubernetes, OpenShift, vSphere."},
+			{Name: "created_at", Type: proto.ColumnType_TIMESTAMP, Description: "The time when the rule was created."},
+			{Name: "built_in", Type: proto.ColumnType_BOOL, Description: "Indicates whether the rule is built-in or custom."},
+			{Name: "description", Type: proto.ColumnType_STRING, Description: "The description of the rule."},
+			{Name: "short_id", Type: proto.ColumnType_STRING, Description: "A short unique identifier for the rule."},
```
Comment (misraved) [inline wiz/table_wiz_cloud_config_rule.go:42]:
```suggestion
			{Name: "short_id", Type: proto.ColumnType_STRING, Description: "A short unique identifier of the rule."},
```

File: wiz/table_wiz_cloud_config_rule.go [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+	// Create client
+	conn, err := getClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("wiz_cloud_config_rule.getCloudConfigRule", "connection_error", err)
+		return nil, err
+	}
+
+	query, err := api.GetCloudConfigRule(context.Background(), conn, id)
+	if err != nil {
```
Comment (misraved) [inline wiz/table_wiz_cloud_config_rule.go:173]: Does the API automatically handle the not found errors?

File: wiz/table_wiz_issue.go [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+	// Return nil, if empty
+	if id == "" {
+		return nil, nil
+	}
+
+	// Create client
+	conn, err := getClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("wiz_issue.getIssue", "connection_error", err)
```
Comment (misraved) [inline wiz/table_wiz_issue.go:160]:
```suggestion
		plugin.Logger(ctx).Error("wiz_issue.getWizIssue", "connection_error", err)
```

File: wiz/table_wiz_issue.go [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+		return nil, err
+	}
+
+	query, err := api.GetIssue(context.Background(), conn, id)
+	if err != nil {
+		plugin.Logger(ctx).Error("wiz_issue.getIssue", "query_error", err)
```
Comment (misraved) [inline wiz/table_wiz_issue.go:166]:
```suggestion
		plugin.Logger(ctx).Error("wiz_issue.getWizIssue", "query_error", err)
```

File: wiz/table_wiz_project.go [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+		Get: &plugin.GetConfig{
+			Hydrate:    getWizProject,
+			KeyColumns: plugin.SingleColumn("id"),
+		},
+		Columns: []*plugin.Column{
+			{Name: "name", Type: proto.ColumnType_STRING, Description: "The name of the project."},
+			{Name: "id", Type: proto.ColumnType_STRING, Description: "A unique identifier of the project."},
+			{Name: "slug", Type: proto.ColumnType_STRING, Description: "The project slug."},
+			{Name: "description", Type: proto.ColumnType_STRING, Description: "A human-readable description of the project."},
+			{Name: "archived", Type: proto.ColumnType_BOOL, Description: "If tru, the project was archived."},
```
Comment (misraved) [inline wiz/table_wiz_project.go:29]:
```suggestion
			{Name: "archived", Type: proto.ColumnType_BOOL, Description: "If true, the project was archived."},
```

File: wiz/table_wiz_project.go [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+			{Name: "business_unit", Type: proto.ColumnType_STRING, Description: "The project business unit."},
+			{Name: "security_score", Type: proto.ColumnType_INT, Description: "Security score is based on the number of successful assessments ran on this project out of the total assessments."},
```
Comment (misraved) [inline wiz/table_wiz_project.go:31]:
```suggestion
			{Name: "security_score", Type: proto.ColumnType_INT, Description: "Security score is based on the number of successful assessments that ran on this project out of the total assessments."},
```

File: wiz/table_wiz_project.go [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+			{Name: "cloud_account_count", Type: proto.ColumnType_INT, Description: "The count of cloud account associated with the project."},
+			{Name: "cloud_organization_count", Type: proto.ColumnType_INT, Description: "The count of cloud organization associated with the project."},
```
Comment (misraved) [inline wiz/table_wiz_project.go:33]:
```suggestion
			{Name: "cloud_organization_count", Type: proto.ColumnType_INT, Description: "The count of cloud organizations associated with the project."},
```

File: wiz/table_wiz_security_category.go [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+		Get: &plugin.GetConfig{
+			Hydrate:    getSecurityCategory,
+			KeyColumns: plugin.SingleColumn("id"),
+		},
+		Columns: []*plugin.Column{
+			{Name: "name", Type: proto.ColumnType_STRING, Description: "The name of the category."},
+			{Name: "id", Type: proto.ColumnType_STRING, Description: "A unique identifier of the category."},
+			{Name: "description", Type: proto.ColumnType_STRING, Description: "The description of the category."},
+			{Name: "framework_id", Type: proto.ColumnType_STRING, Description: "The ID of security framework, the category is part of.", Transform: transform.FromField("Framework.Id")},
+			{Name: "sub_categories", Type: proto.ColumnType_JSON, Description: "A list of security category."},
```
Comment (misraved) [inline wiz/table_wiz_security_category.go:33]:
```suggestion
			{Name: "sub_categories", Type: proto.ColumnType_JSON, Description: "A list of security categories."},
```

File: wiz/table_wiz_security_framework.go [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+			Hydrate:    getSecurityFramework,
+			KeyColumns: plugin.SingleColumn("id"),
+		},
+		Columns: []*plugin.Column{
+			{Name: "name", Type: proto.ColumnType_STRING, Description: "The name of the security framework."},
+			{Name: "id", Type: proto.ColumnType_STRING, Description: "A unique identifier of the security framework."},
+			{Name: "enabled", Type: proto.ColumnType_BOOL, Description: "If true, the security framework is enabled."},
+			{Name: "built_in", Type: proto.ColumnType_BOOL, Description: "If true, the security framework is managed by Wiz."},
+			{Name: "description", Type: proto.ColumnType_STRING, Description: "The description of the security framework."},
+			{Name: "policy_types", Type: proto.ColumnType_JSON, Description: "A list of security framework policy type."},
```
Comment (misraved) [inline wiz/table_wiz_security_framework.go:34]: Does the `policy_types` column need no transformation? Can we query its components like any other IAM policy?

File: wiz/table_wiz_subscription.go [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+			KeyColumns: plugin.SingleColumn("id"),
+		},
+		Columns: []*plugin.Column{
+			{Name: "name", Type: proto.ColumnType_STRING, Description: "The display name for the account."},
+			{Name: "id", Type: proto.ColumnType_STRING, Description: "A unique identifier of the account."},
+			{Name: "cloud_provider", Type: proto.ColumnType_STRING, Description: "The type of the cloud provider. Possible values are: AWS, GCP, OCI, Alibaba, Azure, Kubernetes, OpenShift, vSphere."},
+			{Name: "status", Type: proto.ColumnType_STRING, Description: "Cloud Account connectivity status as affected by configured connectors.  Possible values: CONNECTED, DISABLED, DISCONNECTED, DISCOVERED, ERROR, INITIAL_SCANNING, PARTIALLY_CONNECTED."},
+			{Name: "last_scanned_at", Type: proto.ColumnType_TIMESTAMP, Description: "The time when the cloud account was last scanned."},
+			{Name: "container_count", Type: proto.ColumnType_INT, Description: "Number of containers that are part of this cloud account."},
+			{Name: "external_id", Type: proto.ColumnType_STRING, Description: "External subscription id from cloud provider (subscriptionId in security graph)."},
```
Comment (misraved) [inline wiz/table_wiz_subscription.go:35]:
```suggestion
			{Name: "external_id", Type: proto.ColumnType_STRING, Description: "External subscription ID from a cloud provider (subscriptionId in security graph)."},
```

**Review by misraved — COMMENTED** [review]
@Subhajit97 Please take a look at the minor review comments. Thanks!!

A couple of questions to also be considered:
- Does the API automatically return nil for invalid resources (when an invalid id is passed in the where clause)?
-Is the API prone to throttling? In that case, should we define retriable errors?

File: wiz/table_wiz_security_framework.go (hunk 12 above)
Comment (Subhajit97) [reply wiz/table_wiz_security_framework.go:34]: @misraved the policy types returns an array of policy types that indicates the type of resource to which a security policy applies.

**Comment by misraved** [comment]
@Subhajit97 could you please remove references to APIToken?

**Review by misraved — APPROVED** [review]

**Review by misraved — APPROVED** [review]

**Pull request merged**